lib/hawk/model/association.rb

```diff
@@ -79,6 +79,17 @@ def inherited(subclass)
             @_associations ||= {}
 
             parent.associations.each do |name, (type, options)|
+              self_constant   = '::' + [ self.to_s.deconstantize.presence,   name.to_s.singularize.classify].compact.join('::')
+              parent_constant = '::' + [ parent.to_s.deconstantize.presence, name.to_s.singularize.classify].compact.join('::')
+
+              namespaced = Object.const_get(self_constant) rescue nil
+
+              options[:class_name] = if namespaced && (parent.to_s.deconstantize != self.to_s.deconstantize)
+                self_constant
+              else
+                parent_constant
+              end
+
               _define_association(name, type, options)
```

Comment:
Here we could use some code comments, such as *The idea is to look up this class' name in the same namespace*